site/content/blog/2021-11-30-the-eval-villain-add-on/index.md

```diff
@@ -0,0 +1,105 @@
+---
+title: "The Eval Villain Add-on"
+summary: >
+  Eval Villain was recently added to the ZAP Marketplace.
+  This add-on installs the Eval Villain web extension in Firefox and allows the inspection of arguments to arbitrary native JavaScript functions.
+type: post
+tags:
+- blog
+- browser
+- firefox
+date: "2021-11-30"
+authors: 
+    - bemodtwz
+---
+Eval Villain was recently added to the ZAP Marketplace - this add-on installs the 
+[Eval Villain web extension](https://addons.mozilla.org/en-GB/firefox/addon/eval-villain/) when you 
+[launch Firefox](/blog/2021-11-26-launching-browsers-with-extensions/).
+
+This extension allows the inspection of arguments to arbitrary native JavaScript functions. This tool has similar use cases to DOM Invader in Burp— but Eval Villain has a *lot* more configuration options. As the author of Eval Villain, I was invited to do this guest blog post and explain how to get the most out of the tool.
+
+## Eval Villain Quick Start 
+
+To get started with Eval Villain launch Firefox from ZAP, select the Eval Villain icon on the top right hand side of Firefox, click the enable switch and reload the current web page. Now open the console and you should see a message at the top like "[EV] Functions hooked for <<url>>". This means Eval Villain is loaded and is working for the current URL. If there are DOM interactions you may see more in the page, and if you see other messages in red you may have already found DOM XSS!
```

Comment:
Is this referring to the devtools console or something in Eval Villain?

Reply:
Dev tools console. That is a good point, readers don't know that Eval Villain does not have a console. So maybe it would be good to make it more explicit. Like "open the developer tools and go to the console there you should see...".

Reply:
Dev tools console. That is a good point, readers don't know that Eval Villain does not have a console. So maybe it would be good to make it more explicit. Like "open the developer tools and go to the console there you should see...".